src/PerfView/PerfViewData.cs

```diff
@@ -5713,6 +5713,10 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                 var heapParser = new HeapTraceProviderTraceEventParser(eventSource);
                 Dictionary<Address, StackSourceSample> lastHeapAllocs = null;
 
+                Dictionary<TraceModuleFile, NativeSymbolModule> loadedModules = new Dictionary<TraceModuleFile, NativeSymbolModule>();
```

Comment:
I won't require this, but you can turn loadedModules into an array of NativeSymbolModule.   Every TraceModuleFile has a Index property which is a small integer whose bound is given by TraceModuleFiles.Count.   Thus you can turn the dictionary into an array.

Reply:
It's a bit awkward to keep track of initialization with an array. Would you use something like Optional<NativeSymbolModule>[] or (NativeSymbolModule, bool)[]?